RELEASE/scripts/autoscend.ash

```diff
@@ -2384,20 +2384,140 @@ boolean powerLevelAdjustment()
 	return false;
 }
 
+boolean unlockThinknerdWarehouse()
+{
+	//returns true if you have access to [The Thinknerd Warehouse], if you don't have access tries to unlock it and then return true.
+	//returns false if fails to unlock.
+	//much easier to do if you already have torso awaregness
+	
+	if(internalQuestStatus("questM22Shirt") > -1)
+	{
+		return true;
+	}
+	
+	//unlocking is a multi step process. We want to try things in reverse to conserve resources and in case some steps were already complete.
+	
+	boolean useLetter()
+	{
+		if(item_amount($item[Letter for Melvign the Gnome]) > 0)
+		{
+			if(use(1, $item[Letter for Melvign the Gnome]))
+			{
+				return true;
+			}
+			else
+			{
+				abort("Somehow failed to use [Letter for Melvign the Gnome]... aborting to prevent infinite loops");
+			}
+		}
+		return false;
+	}
+	
+	item target_shirt = $item[none];		//needs to be global so we can call it outside boolean hasShirt()
+	boolean hasShirt()
```

Comment:
this seems a bit inefficient but no big deal

Reply:
change made